### PR TITLE
[TTAHUB-1783/1802/1803/1804] Session report form

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
- Add session report form & hamburger menu item to access it
- Add new model for files associated with session
- Many UI tests for coverage increase

## How to test
- Select "Create a new session" from the hamburger menu next to a training report on the training reports landing page
- Fill out the form

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1783
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1802
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1803
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1804

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by kryswisnaskas** [comment]
Looks good! 👍 

Overall, I was able to fill out the session report form.

I noticed that in a specific case when the session summary form is modified from complete to incomplete, the navigator will still show "Complete". I am assuming this issue is not related to this PR.

**Review by AdamAdHocTeam — COMMENTED** [review]
Editing an existing event via the context menu I get this:

image

File: src/migrations/20230614164742-create-session-report-pilot-files.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      await queryInterface.createTable('SessionReportPilotFiles', {
```
Comment (AdamAdHocTeam) [inline src/migrations/20230614164742-create-session-report-pilot-files.js:4]: So do we need to do anything here with audit?

**Review by AdamAdHocTeam — COMMENTED** [review]
There are a ton of changes in this PR. Overall looks good 😄. 

I left some minor comments also see @kryswisnaskas comments.

UI testing went well I was able to create and update existing session reports.

File: frontend/src/pages/SessionForm/index.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+  const [error, setError] = useState();
+
+  // this is the error that appears in the sidebar
+  const [errorMessage, updateErrorMessage] = useState();
+
+  /* ============ */
+
+  // this attempts to track whether or not we're online
+  // (or at least, if the backend is responding)
+  const [connectionActive] = useState(true);
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply frontend/src/pages/SessionForm/index.js:77]: I'll remove it for now

**Comment by thewatermethod** [comment]
> I noticed that in a specific case when the session summary form is modified from complete to incomplete, the navigator will still show "Complete". I am assuming this issue is not related to this PR.

@kryswisnaskas 

I am not seeing this issue generally: 

https://github.com/HHS/Head-Start-TTADP/assets/3288586/6a891ea1-960e-4d4e-8364-1a0d7dc8c054

Do you know if it is a specific field or am I not triggering what you are seeing the right way?

**Review by AdamAdHocTeam — APPROVED** [review]
Overall LGTM.

**Comment by thewatermethod** [comment]
@AdamAdHocTeam & @kryswisnaskas 

I am going to merge this into main today, unless there are objections. Will try to wrap everything else with permissions as I create the "View event" page

**Pull request merged**